src/pip/_internal/vcs/__init__.py

```diff
@@ -133,29 +133,23 @@ def unregister(self, cls=None, name=None):
         else:
             logger.warning('Cannot unregister because no class or name given')
 
-    def get_backend_name(self, location):
+    def get_backend_type(self, location):
```

Comment:
Any preferences for changing this from returning the name to type?

Reply:
The reason I'm making this change is that this method is used in only one place in the entire code base, and that's the `_init_args_from_dist()` method being modified in this PR. The type return value is exactly what is needed. If it returned the name, it would need to get converted back to the type before it could be used.

Basically, this results in the simplest and least amount of code. Further simplifications that build on this I'll be making in subsequent PR's.

Reply:
Thank you for reviewing! 👍I'll post the subsequent PR soon.